deploy/common/k8sbased.sh

```diff
@@ -323,18 +330,28 @@ function launch_central {
         helm lint "$unzip_dir/chart" -n stackrox "${helm_args[@]}"
       fi
 
-      # set custom central version if the CENTRAL_CHART_VERSION env is set
+      local helm_chart="$unzip_dir/chart"
+      root_dir
+      local latest_tag=$(make tag -C "${STACKROX_ROOT_DIR}" --quiet --no-print-directory)
+
+      # set custom central version if the CENTRAL_CHART_VERSION_OVERRIDE env is set and not equal to the latest tag
       # this has to happen after linting since helm lint does not recognize the --version flag
-      if [[ ! -z ${CENTRAL_CHART_VERSION} ]]; then
+      if [[ ( -n "${CENTRAL_CHART_VERSION_OVERRIDE}" ) && "${CENTRAL_CHART_VERSION_OVERRIDE}"!="${latest_tag}" ]]; then
         helm_args+=(
-          --version="${CENTRAL_CHART_VERSION}"
+          --version="${CENTRAL_CHART_VERSION_OVERRIDE}"
         )
+        # make sure the helm chart is available on the cluster
+        helm repo add stackrox-oss https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource
+        helm repo update
+        helm search repo stackrox-oss -l
+        helm_chart="stackrox-oss/stackrox-central-services"
```

Comment:
@jschnath Seems like you haven't fully read this my comment https://github.com/stackrox/stackrox/pull/5795#discussion_r1206693059

Related part is:

> Before going the path of also overriding chart directory with repo indicator, I'd warn that we have multiple Helm repos and we don't necessarily always take charts from [stackrox/helm-charts@main/opensource](https://github.com/stackrox/helm-charts/tree/main/opensource?rgh-link-date=2023-05-26T12%3A11%3A17Z). Also, it's not necessary that helm has all repositories already configured and have expected names (remember: this script runs both in CI and on developer machines).
Let's what is the right place to override the chart version (disclaimer: I don't have an answer).

I do think that using opensource helm charts unconditionally in this script is wrong because it is not universal. There are "rhacs" and "development_build" helm charts that one might want to use but we take away that option.

Reply:
Would it make sense then to add variables `HELM_CHART_OVERRIDE_NAME` and `HELM_CHART_OVERRIDE_URL` which set the name of the repo and URL? I could then set those to `stackrox-oss` and `https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource` for the compatibility tests and they could be set to `rhacs` and `development_build`in other contexts.

Reply:
The approach has changed (to the one I like :stuck_out_tongue:) so resolving this thread as outdated.